src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
Yeah that would have been my question as well. Personally I would go with the first approach, but then we might be hiding a logic error.